src/shared/components/ProfilePage/Stats/DistributionGraph/index.jsx

```diff
@@ -0,0 +1,288 @@
+/* eslint-env browser */
+import d3 from 'd3';
+import React from 'react';
+import { toPairs } from 'lodash';
+import PT from 'prop-types';
+import ChartTooltip from '../ChartTooltip';
+import './index.scss';
+
+const getRanges = distribution => (
+  toPairs(distribution)
+    .map(([name, number]) => ({
+      name,
+      number,
+      start: parseInt(name.match(/\d+/)[0], 10),
+      end: parseInt(name.match(/To(\d+)/)[1], 10),
+    }))
+    .sort((a, b) => a.start - b.start)
+);
+
+function removeLeadingZeroes(ranges) {
+  while (ranges[0].number === 0) {
+    ranges.shift();
+  }
+}
+
+function removeTrailingZeroes(ranges) {
+  while (ranges[ranges.length - 1].number === 0) {
+    ranges.pop();
+  }
+}
+
+function ratingToColor(colors, rating) {
+  const filteredColors = colors.filter(color => rating >= color.start && rating <= color.end);
+  return (filteredColors[0] && filteredColors[0].color) || 'black';
+}
```

Comment:
We already have functions to convert rating to rating bucket, and style mixins for setting resulting colors in the uniform way everywhere. Check https://github.com/topcoder-platform/community-app/blob/new-develop/src/shared/utils/tc.js#L74-L120 and any code that uses it.